app/models/course/discussion/post.rb

```diff
@@ -1,5 +1,18 @@
 class Course::Discussion::Post < ActiveRecord::Base
   acts_as_forest order: :created_at
 
+  after_initialize :set_topic, if: :new_record?
+  validate :parent_topic_consistency
+
   belongs_to :topic, inverse_of: :posts
+
+  scope :ordered_by_created_at, -> { order(created_at: :asc).includes(:creator).to_a }
+
+  def set_topic
+    self.topic = parent.topic if parent
```

Comment:
would this be better written as `self.topic ||= parent.topic if parent`? This is a default. The validation for parent consistency will come later.